src/App.jsx

```diff
@@ -1,87 +1,98 @@
-      //Number button
-      setDisplayValue(
-        displayValue === "0" ? buttonValue : displayValue + buttonValue
-      );
-    } else if (operators.includes(buttonValue)) {
-      //Operator button
-      setPreviousValue(displayValue);
-      setOperator(buttonValue);
-      setDisplayValue(operator.toString());
-    } else if (buttonValue === "C") {
-      //Clear button
-      setDisplayValue("0");
-      setPreviousValue("");
-      setOperator("");
-    } else if (buttonValue === "=") {
-      //Equals button
-      const num1 = Number(previousValue);
-      const num2 = Number(displayValue);
-      let result;
-
-      if (operator === "+") {
-        result = num1 + num2;
-      } else if (operator === "-") {
-        result = num1 - num2;
-      } else if (operator === "*") {
-        result = num1 * num2;
-      } else if (operator === "/") {
-        result = num2 === 0 ? "Error" : num1 / num2;
+  const handleButtonPress = ({ value, type }) => {
+    if (display.length < maxDisplayLength) {
+      if (type === "operator" && display === "0") {
+        setDisplay(value);
+      } else if (type === "number") {
+        setDisplay(display === "0" ? value : display + value);
+      } else if (type === "operator") {
+        if (!isNaN(lastChar)) {
+          setDisplay(display + value);
+        } else if (lastChar && type === "operator") {
+          setDisplay(display.slice(0, -1) + value);
+        }
+      } else if (type === "decimal") {
+        if (lastChar === ".") {
+          return;
+        } else if (isNaN(lastChar)) {
+          setDisplay(display + "0" + value);
+        } else {
+          setDisplay(display + value);
+        }
       }
+    }
+  };
 
-      if (result !== undefined) {
-        setDisplayValue(result.toString());
-        setShowResult(true);
-        setPreviousValue("");
-        setOperator("");
-      } else {
+  const handleActions = ({ value, type }) => {
+    if (type === "action") {
+      if (value === "AC") {
+        setDisplay("0");
         setShowResult(false);
+      } else if (value === "C") {
+        if (display === "0") {
+          return;
+        } else if (display.length === 1) {
+          setDisplay("0");
+        } else {
+          setDisplay(display.slice(0, -1));
+        }
+      } else if (value === "=") {
+        {
+          setDisplay(display === "0" ? "0" : eval(display).toString());
+          setShowResult(true);
+        }
       }
     }
   };
 
+  const handleClicks = ({ value, type }) => {
+    if (type === "action") {
+      handleActions({ value, type });
+    } else {
+      handleButtonPress({ value, type });
+    }
+  };
+
   return (
     <div className="bg-slate-900 min-h-screen text-white flex flex-col items-center justify-center">
       <div className="min-w-[320px] bg-black flex flex-col gap-4 p-4 rounded-2xl">
         <div className="overflow-x-auto bg-[#282828] text-3xl min-h-[80px] flex items-end justify-center flex-col p-4 rounded-[10px]">
```

Comment:
min-h-[80px] is the min-h-20 class of tailwind

For this application there's no problem, but for a larger application it's better if the custom colors and border radius are specified in the tailwind.config file using the extend setting.

Reply:
I don't know how to do this. But I will take a look at it. I'm still learning tailwind, but up until now I like it very much. thanks

Reply:
You need to do something like https://tailwindcss.com/docs/theme; you only need the “extend” part, as it keeps the tailwind settings and extends them with your own.